WooCommerce/Classes/ViewRelated/Orders/Order Details/Issue Refunds/IssueRefundViewModel.swift

```diff
@@ -94,6 +94,16 @@ final class IssueRefundViewModel {
         return resultsController.fetchedObjects.first
     }()
 
+    /// Charge related to the order. Used to show card details in the `Refund Via` section.
+    ///
+    private lazy var charge: WCPayCharge? = {
+        guard let resultsController = createWcPayChargeResultsController() else {
```

Comment:
This is a bit limiting – it's optional because `chargeID` on Order is necessarily optional, and sometimes we don't have the `chargeID` in time when starting a refund. In that case, we're never able to fetch the charge. I'll look to recreate the `resultsController` when the `chargeID` changes, I think.

Reply:
Maybe we could introduce Combine to `EntityListener`? I imagine if we had that we could observe the order for changes, then `flatMap` that to observe the charge. Completely untested rough example 🙈:

https://gist.github.com/koke/1a6cef8f1a0012719cd757875542a4df